ansible/roles/jupyterhub/files/jupyterhub_config.py

```diff
@@ -240,3 +240,21 @@
 ##########################################
 # END CUSTOM DOCKERSPAWNER
 ##########################################
+
+# Dynamic config to setup new courses
+
+# course setup service name
+service_name = os.environ.get('DOCKER_SETUP_COURSE_SERVICE_NAME', 'setup-course')
+
+# course setup service port
+port = os.environ.get('DOCKER_SETUP_COURSE_PORT', '8000')
+
+# get the response from course setup app endpoint
+response = requests.get(f'http://{service_name}:{port}/config')
```

Comment:
Let's use the utils method to fetch the origin scheme (leftmost value in x-forwarded-proto), so we can support TSL for all internal communications.

Reply:
I understand your point, but the use of `get_client_protocol`  method (from 'illumidesk.authenticators.utils') requires a RequestHandler instance and in this config file we are not using it. 
`def get_client_protocol(self, handler: RequestHandler) -> Dict[str, str]: ...`

Reply:
@jgwerner  what do you think?

Reply:
@netoisc good catch. In that case perhaps we need to create a new abstraction where we can deal with requests from specific frameworks when implementing the method, but we should probably do that with a new enhancement PR.